Updated User Management, RBAC and API Ownership (#4236)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/user-management/api-ownership.md

{{< note success >}}
**Note**  

The availability of this feature [depends on your licence]({{< ref "tyk-dashboard/rbac" >}}).
{{< /note >}} 

## When to use API Ownership
#### Governing a multi-team API portfolio
API ownership is a key capability when you have multiple API development teams each working on their own suite of APIs. You can use API Ownership to simplify the experience of those developers when accessing Tyk by reducing the "clutter" of APIs that they are not working with, and also to avoid the risk of users accidentally or maliciously interfering with another team's APIs.

#### Avoiding data leakage between users
The [user owned analytics]({{< ref "product-stack/tyk-dashboard/advanced-configurations/user-management/api-ownership#owned-analytics" >}}) feature allows you to prevent users from seeing the traffic to (and existence of) APIs for which they are not responsible. This reduces the opportunity for data leakage within your business.

## How API Ownership works

> A Dashboard User (or User Group) can be assigned as *owner* of an API, granting them visibility of and - given appropriate permissions - the ability to manage all the Tyk objects relating to that API, such as policies, key requests and Dashboard analytics. Assigning an owner to an API restricts visibility of that API and its related objects to other users who will not be able to view or manage it.

Author: ```suggestion
By default, APIs and associated objects (such as policies and Dashboard analytics) are visible to all Tyk Dashboard users.

A Dashboard User (or User Group) can be assigned as the *owner* of an API, granting that user (or user group) exclusive visibility of and - given appropriate permissions - the ability to manage all the Tyk objects relating to that API, such as policies, key requests and Dashboard analytics.

Once an API has been assigned an *owner*, all non-owning users will lose visibility of - and access to - that API in the Tyk Dashboard.
```

> 

Author: ```suggestion
```

> When there is no owner assigned to an API, it and objects using it are visible to all users.

Author: ```suggestion
```

You can remove owners from the API by clicking on the `x` next to their name in the drop-down/selection box.

#### Using Tyk Classic APIs
When using Tyk Classic APIs, the option to assign owner(s) to an API is provided on the **Core Settings** tab in the API Designer. You simply select the owner (or owners) that you wish to assign to the API from the drop-down boxes:
{{< img src="/img/dashboard/system-management/api_ownership.png" alt="API ownership section for Tyk Classic APIs" >}}

You can remove owners from the API by deselecting them from the drop-down.

## Managing API owners using the Dashboard API
The [Tyk Dashboard API]({{< ref "tyk-dashboard-api" >}}) provides endpoints to manage API ownership directly, if you are not using the API Designer.

#### Using Tyk OAS APIs
When working with Tyk OAS APIs, you can manage owners for an API using these endpoints:

| Method | Endpoint path           | Action                                                                                 |
|--------|-------------------------|----------------------------------------------------------------------------------------|

> | `PUT`  | `/apis/{apiId}/access`  | Assign a list of owners to the specified API                                           |

Author: Typo

| `GET`  | `/apis/{apiID}/access`  | Retrieve the list of owners of the specified API                                       |

For each of these endpoints, the payload consists of two string lists: one for user IDs, the other for user group IDs.
``` json
{
  "userIds": [
    "string"
  ],
  "userGroupIds": [
    "string"
  ]
}
```

#### Using Tyk Classic APIs
When working with Tyk Classic APIs, you manage owners for an API by modifying the `user_owners` and `user_group_owners` fields in the API definition and then updating the API in Tyk with that using these endpoints:

| Method | Endpoint            | Action                                                                                                                |
|--------|---------------------|-----------------------------------------------------------------------------------------------------------------------|

> | `PUT`  | `/api/apis/{apiId}` | Update the API definition for the specified API - CRUD API owners in the `user_owners` and `user_group_owners` fields |

Author: Typo
